crates/proto/build.rs

```diff
@@ -8,24 +8,24 @@ use prost::Message;
 /// This is done only if BUILD_PROTO environment variable is set to `1` to avoid running the script
 /// on crates.io where repo-level .proto files are not available.
 fn main() -> miette::Result<()> {
-    println!("cargo:rerun-if-changed=generated");
-    println!("cargo:rerun-if-changed=../../proto");
+    let crate_root = PathBuf::from_str(std::env!("CARGO_MANIFEST_DIR")).into_diagnostic()?;
+    let workspace_root =
+        crate_root.parent().and_then(|crates| crates.parent()).ok_or_else(|| {
+            miette::miette!("Failed to navigate two directories up into workspace root")
+        })?;
```

Comment:
Would this still work when we publish to crates.io? (there may not be a workspace root there).

One potential option is to move this code after we check for `BUILD_PROTO` environment variable. But I'm not sure how it will affect things.

Reply:
Ah I see and it worked before because the publish abort check was earlier. I'll see if I can't just move the check to the beginning.

Alternative thought: lets just remove the build script entirely. We already have the generated files checked into git. So really its only buying us the convenience of not having to think about things when we fiddle with the proto files (which should be fairly rare). And if you're changing the proto files you can just manually run protoc (which we can simply add as a script file for ease of use, or to the readme as an instruction). For safety we can add a CI step that checks that protoc output still matches the generated files.

Reply:
If not too difficult, I think the alternative approach would be preferred. Specifically:

- Add something like `proto` command (and an associated script) to the Makefile so that we could run `make proto` to rebuild the generated file.
- Add a CI check to make sure that if any of the `.proto` files changed, the generated files are up to date.

One comment though: if we go this route, we should also make sure we can do the same for the [build script](https://github.com/0xPolygonMiden/miden-node/blob/main/crates/rpc-proto/build.rs) in the `rpc-proto` crate.

Reply:
Who/what uses the `write_proto` function?

Reply:
https://github.com/0xPolygonMiden/miden-client/blob/main/crates/rust-client/build.rs#L21

Reply:
I've updated the description for my new attempt. Might be a bit too extreme of a change though it does seem to work.